### PR TITLE
[FLINK-2591] Add configuration parameter for default number of yarn containers

### PR DESCRIPTION
**Pull request by willmiao**

### CODE REVIEW

**Comment by rmetzger** [comment]
Sorry for the late response, I didn't see this new pull request.

The failed test is okay. The test is known for being unstable.

File: flink-dist/src/main/resources/flink-conf.yaml [hunk 1]
```diff
@@ -130,6 +130,15 @@ state.backend: jobmanager
 
 
 #==============================================================================
+# YARN
+#==============================================================================
+
+# Default number of YARN container to allocate (=Number of Task Managers)
+
+yarn.defaultNumberOfTaskManagers: 1
```
Comment (rmetzger) [inline flink-dist/src/main/resources/flink-conf.yaml]: Can you remove the configuration value from the default configuration again?

I would like to force new users to specify the number of yarn containers when they start Flink on YARN.
The configuration value is meant for production users which want to control everything using configuration values.

File: flink-yarn-tests/src/main/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java [hunk 2]
```diff
@@ -111,6 +111,20 @@ public void testClientStartup() {
 	}
 
 	/**
+	 * Test configuration parameter for default number of yarn containers
+	 */
+	@Test
+	public void testDefaultNumberOfTaskManagers() {
+		LOG.info("Starting testDefaultNumberOfTaskManagers()");
```
Comment (rmetzger) [inline flink-yarn-tests/src/main/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java]: I would generate a flink-conf.yaml just for this test to test the behavior

**Comment by rmetzger** [comment]
For others who follow this PR, there was already some discussion on this change here: https://github.com/apache/flink/pull/1107

**Comment by rmetzger** [comment]
@willmiao: What is the status of this pull request?

**Comment by willmiao** [comment]
@rmetzger : I'm sorry for the late reply, I have been on vacation last three weeks.
I failed to find a nice way to "generate a flink-conf.yaml just for the test", could you please give some tips on this?

**Comment by rmetzger** [comment]
That's indeed a good question. I don't have much time currently, but I'm trying to answer you within the next few days.

**Comment by willmiao** [comment]
Hi rmetzger,
I have updated this PR.
I generated a flink-conf.yaml in a directory, which I then used as the configuration directory for the test.

**Comment by willmiao** [comment]
Sorry for the failed tests, I'll fix them.

**Comment by willmiao** [comment]
hi @rmetzger ,
I believe I find the reason why my test failed.
A test will fail if we don’t specify “-t” argument(_which is an optional argument_) in the command when call runWithArgs function, and we will get exception here:
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/log4j/Level
    at org.apache.hadoop.mapred.JobConf.<clinit>(JobConf.java:357)
```
It seems some jar files relevant to log callback are needed.

One more thing, I noticed a segment of code as follows in the file “FlinkYarnSessionCli.java”:
```
File logback = new File(confDirPath + File.pathSeparator + CONFIG_FILE_LOGBACK_NAME);
if (logback.exists()) {
shipFiles.add(logback);
    flinkYarnClient.setFlinkLoggingConfigurationPath(new Path(logback.toURI()));
}
```
And I wonder if “**File.pathSeparator**” should be “**File.separator**” here.

**Pull request closed without merging**